libr/bin/format/elf/elf.c

```diff
@@ -2627,150 +2635,276 @@ char *Elf_(r_bin_elf_get_rpath)(ELFOBJ *bin) {
 	return ret;
 }
 
+static void fix_rva_and_offset(ELFOBJ *bin, RBinElfReloc *r, ut64 offset_table) {
+	if (is_bin_etrel (bin)) {
+		r->rva = offset_table + r->offset;
```

Comment:
where did you get this from? I think it will never be executed as you are calling `fix_rva_and_offset` only from `populate_relocs_record_from_dynamic` and ET_REL objects do not have dynamic segment AFAIK.

Reply:
the if part can be useful if you use `fix_rva_and_offset` in `populate_relocs_record_from_section` as well, but it needs to be changed a bit I think.

Reply:
Indeed i will split this function in two sub function.